tests/test_unit/test_arguments.py

```diff
@@ -98,3 +112,226 @@ def test_argument_parameter_build(arguments, expected_built_arguments):
+            None,
+            id="no-arguments",
+        ),
+        pytest.param(
+            ModelArguments(),
+            ModelArguments(),
+            id="model-arguments",
+        ),
+        pytest.param(
+            [Parameter(name="param-name", value="a-value"), Artifact(name="artifact-name", from_="somewhere")],
+            ModelArguments(
+                parameters=[
+                    ModelParameter(
+                        name="param-name",
+                        value="a-value",
+                    )
+                ],
+                artifacts=[
+                    ModelArtifact(
+                        name="artifact-name",
+                        from_="somewhere",
+                    )
+                ],
+            ),
+            id="mixed-list",
+        ),
+        pytest.param(
+            {"param-name": "a-value", "a-key": ModelArtifact(name="artifact-name", from_="somewhere")},
+            ModelArguments(
+                parameters=[
+                    ModelParameter(
+                        name="param-name",
+                        value="a-value",
+                    )
+                ],
+                artifacts=[
+                    ModelArtifact(
+                        name="a-key",
+                        from_="somewhere",
+                    )
+                ],
+            ),
+            id="mixed-dict",
+        ),
+        pytest.param(
+            [
+                {"param-name": "a-value"},
+                ModelArtifact(name="artifact-name", from_="somewhere"),
+            ],
+            ModelArguments(
+                parameters=[
+                    ModelParameter(
+                        name="param-name",
+                        value="a-value",
+                    )
+                ],
+                artifacts=[
+                    ModelArtifact(
+                        name="artifact-name",
+                        from_="somewhere",
+                    )
+                ],
+            ),
+            id="param-dict-in-list",
+        ),
+        pytest.param(
+            [
+                {"param-name": "a-value"},
+                {"a-key": Artifact(name="artifact-name", from_="somewhere").with_name("ignore-me")},
+            ],
+            ModelArguments(
+                parameters=[
+                    ModelParameter(
+                        name="param-name",
+                        value="a-value",
+                    )
+                ],
+                artifacts=[
+                    ModelArtifact(
+                        name="a-key",
+                        from_="somewhere",
+                    )
+                ],
+            ),
+            id="multiple-dicts-in-list",
+        ),
+    ),
+)
+def test_mixed_arguments_build(arguments, expected_built_arguments):
+    assert (
+        ArgumentsMixin(
+            arguments=arguments,
+        )._build_arguments()
+        == expected_built_arguments
+    )
```

Comment:
nit: Given how simple this test body is, it'd probably be simpler to just have multiple tests.

Reply:
Also unsure what you mean here? I used separate test functions just to help organise

Reply:
I mean splitting `test_mixed_arguments_build` up into multiple, non-parameterized tests.

Reply:
I'd rather use `parametrize` here as they're labelled relatively nicely via `id`, and we've split them based on parameter-only/artifact-only/mixed (`test_argument_parameter_build` is already there)